examples/k3s/composition.yaml

```
@@ -0,0 +1,149 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: k3sclusters.deployments.equinix.com
+spec:
+  writeConnectionSecretsToNamespace: crossplane-system
+  compositeTypeRef:
+    apiVersion: deployments.equinix.com/v1alpha1
+    kind: K3sCluster
+  resources:
+    - name: k3s-project
+      base:
+        apiVersion: metal.equinix.jet.crossplane.io/v1alpha1
+        kind: Project
+      patches:
+        - type: FromCompositeFieldPath
+          fromFieldPath: metadata.name
+          toFieldPath: spec.forProvider.name
+    - name: k3s-userdata
+      base:
+        apiVersion: v1
+        kind: Secret
+        type: Opaque
+        metadata:
+          namespace: crossplane-system
+      patches:
+        - type: CombineFromComposite
+          toFieldPath: metadata.name
+          combine:
+            variables:
+              - fromFieldPath: metadata.name
+            strategy: string
+            string:
+              fmt: "k3s-userdata-%s"
+        - type: FromCompositeFieldPath
+          toFieldPath: metadata.namespace
+          fromFieldPath: spec.secretNamespace
+        - type: CombineFromComposite
+          toFieldPath: data.userdata
```

```suggestion
          toFieldPath: stringData.userdata
```
`data` requires the content to be base64 encoded.